src/wallet/rpcwallet.cpp

```diff
@@ -2575,6 +2577,258 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
+
+    // check that original tx consists entirely of our inputs
+    // if not, we can't bump the fee, because the wallet has no way of knowing the value of the other inputs (thus the fee)
+    if (!pwalletMain->IsAllFromMe(wtx, ISMINE_SPENDABLE))
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Transaction contains inputs that don't belong to this wallet");
+
+    // bounds-check the output index to decrement
+    int nOutput = request.params[1].get_int();
+    if (nOutput < 0 || (unsigned int) nOutput > wtx.vout.size())
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Output index out of bounds");
+
+    // optional parameters
+    int newConfirmTarget = nTxConfirmTarget;
+    CAmount totalFee = 0;
+    CAmount bumpfeeMaxTxFee = 0.01 * COIN;
+    bool fForce = false;
+    if (request.params.size() > 2) {
+        RPCTypeCheck(request.params, boost::assign::list_of(UniValue::VSTR)(UniValue::VNUM)(UniValue::VOBJ));
+        UniValue options = request.params[2];
+        if (options.size() > 3)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Too many optional parameters");
+        RPCTypeCheckObj(options,
+            {
+                {"confTarget", UniValueType(UniValue::VNUM)},
+                {"totalFee", UniValueType(UniValue::VNUM)},
+                {"maxFee", UniValueType(UniValue::VNUM)},
+                {"force", UniValueType(UniValue::VBOOL)},
+            },
+            true, true);
+
+        if (options.exists("confTarget")) {
+            newConfirmTarget = options["confTarget"].get_int();
+            if (newConfirmTarget <= 0) // upper-bound will be checked by estimatefee/smartfee
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid confTarget (cannot be <= 0)");
+        }
+        if (options.exists("maxFee")) {
+            CAmount proposedMaxFee = options["maxFee"].get_int();
+            if (proposedMaxFee <= 0)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid maxFee (cannot be <= 0)");
+            else if (proposedMaxFee > maxTxFee)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid maxFee (cannot be higher than the global maxTxFee)");
+            bumpfeeMaxTxFee = proposedMaxFee;
+        }
+        if (options.exists("totalFee")) {
+            totalFee = options["totalFee"].get_int();
+            if (totalFee <= 0)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid totalFee (cannot be <= 0)");
+            else if (totalFee > bumpfeeMaxTxFee)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid totalFee (cannot be higher than bumpfee's maxFee)");
+        }
+        if (options.exists("force")) {
+            fForce = options["force"].get_bool();
+        }
+    }
+
+    // calculate the old fee and fee-rate
+    CAmount nOldFee = wtx.GetDebit(ISMINE_SPENDABLE) - wtx.GetValueOut();
+    int txSize = (int)GetVirtualTransactionSize((CTransaction)wtx);
```

Comment:
Just use `int64_t`? `int` is only guaranteed to be 15-bit, so it's too small anyway.